src/pages/ChangeLogPage.spec.js

```diff
@@ -0,0 +1,149 @@
+        name: "Testing GitHub API",
+        created_at: "2022-05-24T18:23:20Z",
+        published_at: "2022-05-26T10:02:09Z",
+        body: "### This is a test to verify if GitHub API can fetch releases\r\n",
+    },
+];
+
+describe("Changelog Page", () => {
+    it("Should show Changelog text", () => {
+        changeLogService.fetchReleases.mockImplementationOnce(
+            () =>
+                new Promise((resolve, _) => {
+                    resolve([]);
+                })
+        );
+
+        const wrapper = renderWithStoreAndTheme(
+            <BrowserRouter>
+                <ChangeLogPage />
+            </BrowserRouter>,
+            { theme: AppTheme }
+        );
+
+        expect(wrapper.getByText("Changelog")).toBeInTheDocument();
+    });
+
+    it("Last Updated text should have the most recent date", async () => {
+        changeLogService.fetchReleases.mockImplementationOnce(
+            () =>
+                new Promise((resolve, _) => {
+                    resolve(mockedResponse);
+                })
+        );
+
+        let wrapper;
+        await act(() => {
+            wrapper = renderWithStoreAndTheme(
+                <BrowserRouter>
+                    <ChangeLogPage />
+                </BrowserRouter>,
+                { theme: AppTheme }
+            );
+        });
+
+        expect(wrapper.getByText("Changelog")).toBeInTheDocument();
+
+        const releaseDates = wrapper
+            .getAllByTestId("releaseDate")
+            .map((release) => release.innerHTML);
+
+        let mostRecentDate = null;
+        let mostRecentText = "";
+        for (const releaseDate of releaseDates) {
+            const formattedDate = new Date(
+                `${releaseDate.substring(3, 5)}-${releaseDate.substring(
+                    0,
+                    2
+                )}-${releaseDate.substring(6)}`
+            );
+            if (
+                mostRecentDate === null ||
+                isAfter(formattedDate, mostRecentDate)
+            ) {
+                mostRecentDate = formattedDate;
+                mostRecentText = releaseDate;
+            }
+        }
+
+        expect(wrapper.getByText(`Last updated: ${mostRecentText}`));
+    });
+
+    it("Should show snackbar message", async () => {
+        addSnackbar.mockImplementationOnce(() => ({ type: "" }));
+        changeLogService.fetchReleases.mockImplementationOnce(
+            () =>
+                new Promise((_, reject) => {
+                    reject("Error fetching the releases");
+                })
+        );
+
+        const wrapper = renderWithStoreAndTheme(
+            <BrowserRouter>
+                <ChangeLogPage />
+            </BrowserRouter>,
+            { theme: AppTheme }
+        );
+
+        expect(wrapper.getByText("Changelog")).toBeInTheDocument();
+
+        await waitFor(() => {
+            expect(addSnackbar).toHaveBeenCalledTimes(1);
+        }, {
+            timeout: 1000,
+        });
+    });
```

Comment:
I know this is also used somewhere else but why is the timeout needed for this snackbar but not for the cookie notification, for example? Just want to understand the difference

Reply:
I'm not aware of the cookie notification case, but here I use this waitFor since the addSnackbar method is being called after a certain method (fetchData) catches an error, so it is not called instantly. I'm not 100% sure this is the reason why, but using the `act` method to wait for updates is not enough for this case, I think it is because there isn't a component state update, only an error catching. By using the `waitFor` it will assert multiple times and will finish earlier if the assertion is correct.